service/matching/task_queue_partition_manager.go

```diff
@@ -228,6 +199,12 @@ func (pm *taskQueuePartitionManagerImpl) AddTask(
 		// default queue should stay alive even if requests go to other queues
 		pm.defaultQueue.MarkAlive()
 	}
+
+	if pm.partition.IsRoot() && !pm.HasPollerAfter(time.Now().Add(-noPollerThreshold)) {
```

Comment:
is there a plan for how to count no-pollers for individual build ids? (of course no need to fix it here)